boa/vyper/contract.py

```diff
@@ -585,6 +588,20 @@ def marshal_to_python(self, computation, vyper_typ):
         if computation.is_error:
             self.handle_error(computation)
 
+        # if call profiling is enabled, store gas used for each call:
+        if self.env.profile_calls:
+            try:
+                fn_name = self._get_fn_from_computation(computation).name
+            except AttributeError:
+                # TODO: https://github.com/vyperlang/vyper/issues/3200  # noqa: E501
+                fn_name = "unnamed"
+            fn_name = self.compiler_data.contract_name + "." + fn_name
+            gas_used = computation.get_gas_used()
+            if fn_name not in self.env.profiled_calls.keys():
+                self.env.profiled_calls[fn_name] = [gas_used]
```

Comment:
there is a problem here if two contracts have the same name and have a function with the same name.